File: problems/003/003.f90
PROGRAM F003

    ! Copyright 2021 Melwyn Francis Carlo
    ! FILE REFERENCE: http://www.naturalnumbers.org/primes.html

    IMPLICIT NONE

    INTEGER :: PRIME_NUM
    REAL (KIND=8), PARAMETER :: N = 600851475143.D0
    REAL (KIND=8), PARAMETER :: PRIME_NUM_MAX = SQRT(N)
    REAL (KIND=8) :: PRIME_NUM_LARGEST
    CHARACTER (LEN=50), PARAMETER :: FILENAME = "problems/003/PrimeNumbers_Upto_1000000"

    OPEN (2, FILE = FILENAME, STATUS = 'OLD')

    PRIME_NUM_LARGEST = N
    DO
        READ(2, *, END=10) PRIME_NUM
        IF (PRIME_NUM <= PRIME_NUM_MAX) THEN
            IF (MOD(INT(N, 8), INT(PRIME_NUM, 8)) == 0) THEN
                PRIME_NUM_LARGEST = PRIME_NUM
            END IF
        ELSE
            EXIT
        END IF
    END DO

    10 CLOSE(2)

    PRINT ('(I0)'), INT(PRIME_NUM_LARGEST)

END PROGRAM F003

File: problems/003/003.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

/* Copyright 2021 Melwyn Francis Carlo */
/* File Reference: http://www.naturalnumbers.org/primes.html */

int main()
{
    FILE *fp;
    char prime_num[10];
    long int n = 600851475143;
    const char* FILE_NAME = "problems/003/PrimeNumbers_Upto_1000000";

    fp = fopen(FILE_NAME, "r");

    long int prime_num_max = sqrt(n);
    long int prime_num_largest = n;

    while (fgets(prime_num, 10, fp) != NULL)
    {
        if (atoi(prime_num) <= prime_num_max)
        {
            if ((n % atoi(prime_num)) == 0)
                prime_num_largest = atoi(prime_num);
        }
        else
        {
            break;
        }
    }

    fclose(fp);

    printf("%ld\n", prime_num_largest);

    return 0;
}
